lib/rules/declaration-block-no-duplicate-properties/index.js

```diff
@@ -88,97 +88,60 @@ const rule = (primary, secondaryOptions, context) => {
 
-				if (indexDuplicate !== -1) {
-					const duplicateDecl = decls[indexDuplicate];
-					const duplicateValue = duplicateDecl ? duplicateDecl.value : '';
-					const duplicateImportant = duplicateDecl ? duplicateDecl.important : false;
-
-					if (ignoreDiffValues || ignorePrefixlessSameValues) {
-						// fails if duplicates are not consecutive
-						if (indexDuplicate !== decls.length - 1) {
-							if (context.fix) {
-								if (!important && duplicateImportant) {
-									decl.remove();
-								} else {
-									removePreviousDuplicate(decls, lowerProp);
-								}
-
-								return;
-							}
-
-							report({
-								message: messages.rejected(prop),
-								node: decl,
-								result,
-								ruleName,
-								word: prop,
-							});
-
-							return;
-						}
-
-						if (ignorePrefixlessSameValues) {
-							// fails if values of consecutive, unprefixed duplicates are equal
-							if (vendor.unprefixed(value) !== vendor.unprefixed(duplicateValue)) {
-								if (context.fix) {
-									if (!important && duplicateImportant) {
-										decl.remove();
-									} else {
-										removePreviousDuplicate(decls, lowerProp);
-									}
-
-									return;
-								}
-
-								report({
-									message: messages.rejected(prop),
-									node: decl,
-									result,
-									ruleName,
-									word: prop,
-								});
-
-								return;
-							}
-						}
-
-						// fails if values of consecutive duplicates are equal
-						if (value === duplicateValue) {
-							if (context.fix) {
-								removePreviousDuplicate(decls, lowerProp);
-
-								return;
-							}
-
-							report({
-								message: messages.rejected(prop),
-								node: decl,
-								result,
-								ruleName,
-								word: prop,
-							});
-
-							return;
-						}
+				if (indexDuplicate === -1) {
+					decls.push(decl);
+				}
 
-						return;
+				const duplicateDecl = decls[indexDuplicate];
+
+				if (!duplicateDecl) {
+					return;
+				}
+
+				const duplicateValue = duplicateDecl.value || '';
+				const duplicateImportant = duplicateDecl.important || false;
+				const duplicateIsMoreImportant = !important && duplicateImportant;
+				const duplicatesAreConsecutive = indexDuplicate === decls.length - 1;
+				const unprefixedDuplicatesAreEqual =
+					vendor.unprefixed(value) === vendor.unprefixed(duplicateValue);
+
+				function fixOrReport() {
+					if (!context.fix) {
+						return report({
+							message: messages.rejected(prop),
+							node: decl,
+							result,
+							ruleName,
+							word: prop,
+						});
 					}
 
-					if (ignoreDuplicates && indexDuplicate === decls.length - 1) {
-						return;
+					if (duplicateIsMoreImportant) {
+						return decl.remove();
 					}
 
-					if (context.fix) {
-						if (!important && duplicateImportant) {
-							decl.remove();
-						} else {
-							removePreviousDuplicate(decls, lowerProp);
-						}
+					if (!duplicateDecl) return;
```

Comment:
either this one is not necessary or the one at l97 is
i.e. if we reach l123 `duplicateDecl` is `true` because it's a const

Reply:
If you remove check at l123 `duplicateDecl` at l125 will be of type `Declaration | undefined`.

Reply:
at l97 you have
```
				if (!duplicateDecl) {
					return;
				}
```
the l123 check is called at l133 and l157 which are after l97.
It seems to me that `duplicateDecl` cannot be `undefined` at that stage because `Boolean(undefined) === false` hence it would have returned early.

Reply:
You can clone the branch